collect_app/src/androidTest/java/org/odk/collect/android/support/matchers/CustomMatchers.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2019 Nafundi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.odk.collect.android.support.matchers
+
+import android.view.View
+import android.widget.TextView
+import org.hamcrest.Description
+import org.hamcrest.Matcher
+import org.hamcrest.TypeSafeMatcher
+import org.odk.collect.android.R
+import org.odk.collect.android.widgets.QuestionWidget
+
+/**
+ * Grab bag of Hamcrest matchers.
+ */
+object CustomMatchers {
+    /**
+     * Matches the view at the given index. Useful when several views have the same properties.
+     * https://stackoverflow.com/a/39756832
+     *
+     */
+    @Deprecated("this matcher is stateful and will cause problems if used more than once")
```

Comment:
The problems with this are actually mentioned in a comment on the SO answer, but for some reason the answer has stayed up. I've flagged it, so others don't run into the same problems.

To be clear, I'm not sure if it actually broke anything for us, but while experimenting with different ways of scrolling and clicking I ended up running into problems caused by the state here.